data/scenarios/Testing/836-pathfinding/_836-automatic-waypoint-navigation/rabbit.sw

```diff
@@ -1,16 +1,27 @@
 
-def visitNextWaypoint = \nextWpIdx.
+def visitNextWaypoint : (rec l. Unit + (Int * Int) * l) -> (rec l. Unit + (Int * Int) * l) -> Cmd Unit = \originalList. \remainingList.
+
     emptyHere <- isempty;
     if emptyHere {
       try {
         place "egg";
       } {};
     } {};
     watch down;
-    let nextWaypointQuery = waypoint "wp" nextWpIdx in
-    teleport self $ nextWaypointQuery;
-    wait 1000;
-    visitNextWaypoint $ nextWpIdx + 1;
+
+    // Wrap around
+    let myList = case remainingList (\_. originalList) (\_. remainingList) in
```

Comment:
Ah, I like it, this is a clever way to use the list of waypoints incrementally without having to do an O(n) index operation every time.